src/Helper/Factory/DoctypeFactory.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Mvc\View\Helper\Factory;
+
+use ArrayAccess;
+use Laminas\View\Helper\Doctype;
+use Psr\Container\ContainerInterface;
+
+use function assert;
+use function is_array;
+use function is_string;
+
+final class DoctypeFactory
+{
+    public function __invoke(ContainerInterface $container): Doctype
+    {
+        $helper = new Doctype();
+
+        if (! $container->has('config')) {
+            return $helper;
+        }
+
+        $config = $container->get('config');
+        assert(is_array($config) || $config instanceof ArrayAccess);
+
+        $options = $config['view_manager'] ?? [];
```

Comment:
_Unfortunately, I don't remember if we had discussed this topic before, so please forgive me if I start a repetition here._

Why keep the name of the configuration key if it is no longer the responsibility of the view manager? This would also remove the difference between Mezzio and laminas-mvc:

- https://docs.laminas.dev/laminas-view/helpers/doctype/#usage-in-a-mezzio-application
- https://docs.laminas.dev/laminas-view/helpers/doctype/#usage-in-a-laminas-mvc-application

Reply:
Not sure if we discussed this previously or not. The main reason for maintaining `view_manager` is its historic significance for MVC apps as you've pointed out. There's probably a lot of MVC apps out there configured with `'view_manager' => ['doctype' => 'HTML5']`.

Because this lib is a bridge between MVC and view, isn't it the responsibility of this package to ease/minimise BC breaks for MVC users?

There's probably other stuff that 'belongs' under `view_manager` that I haven't encountered yet - should they all be changed to use `view_helper_config` forcing users to update all config items?

Reply:
I didn't want to insert a BC break, but that maybe both would work. On the other hand, we can easily clean this up with version 4 of laminas-mvc.

> There's probably other stuff that 'belongs' under view_manager that I haven't encountered yet…

Correct, for example:

- `display_not_found_reason`
- `display_exceptions`
- `template_map`
- `template_path_stack`
- `strategies`
- `not_found_template`
- `exception_template`
- `layout`
- and so on: https://docs.laminas.dev/laminas-mvc/services/#viewmanager

Reply:
So which do you think is better - using a consistent key for all view helper related config, i.e. `view_helper_config` and documenting the changes where required in MVC 4.0 or maintaining the existing disparate keys for MVC and Mezzio?

Reply:
Using `view_manager` here so that the user does not have to change anything in the configuration.
This can be changed later. Who knows, maybe there will no longer be a view manager in version 4.0.